app/models/manageiq/providers/vmware/infra_manager/vm/operations/snapshot.rb

```diff
@@ -26,9 +27,12 @@ def raw_remove_snapshot(snapshot_id)
 
       run_command_via_parent(:vm_remove_snapshot, :snMor => snapshot.uid_ems)
     rescue => err
-      create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "remove")
-      if err.to_s.include?('not found')
-        raise MiqException::MiqVmSnapshotError, err.to_s
+      # If the exception is a VimFault then err.message is a VimString
+      error = err.message.to_s
```

Comment:
wondering if something like `error = String.new(err.message)` would make the intent clearer.

Reply:
:+1: